openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+    )
+    podConfig: Optional[PodConfig] = Field(
+        None, description='Pod specific configuration'
+    )
+    supplementalGroups: Optional[List[int]] = Field(
+        None,
+        description='supplementalGroups defines the linux groups to be applied to the Restic Pod',
+    )
+    timeout: Optional[str] = Field(
+        None, description='timeout defines the Restic timeout, default value is 1h'
+    )
+
+
+class CustomPlugin(BaseModel):
+    image: str
+    name: str
+
+
+class LogLevel(Enum):
+    trace = 'trace'
+    debug = 'debug'
+    info = 'info'
+    warning = 'warning'
+    error = 'error'
+    fatal = 'fatal'
+    panic = 'panic'
+
+
+class TolerationModel(BaseModel):
+    effect: Optional[str] = Field(
+        None,
+        description='Effect indicates the taint effect to match. Empty means match all taint effects. When specified, allowed values are NoSchedule, PreferNoSchedule and NoExecute.',
+    )
+    key: Optional[str] = Field(
+        None,
+        description='Key is the taint key that the toleration applies to. Empty means match all taint keys. If the key is empty, operator must be Exists; this combination means to match all values and all keys.',
+    )
+    operator: Optional[str] = Field(
+        None,
+        description="Operator represents a key's relationship to the value. Valid operators are Exists and Equal. Defaults to Equal. Exists is equivalent to wildcard for value, so that a pod can tolerate all taints of a particular category.",
+    )
+    tolerationSeconds: Optional[int] = Field(
+        None,
+        description='TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute, otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict). Zero and negative values will be treated as 0 (evict immediately) by the system.',
+    )
+    value: Optional[str] = Field(
+        None,
+        description='Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty, otherwise just a regular string.',
+    )
+
+
+class DataProtectionApplicationConfigurationVeleroPodConfig(BaseModel):
+    labels: Optional[Dict[str, str]] = Field(None, description='labels to add to pods')
+    nodeSelector: Optional[Dict[str, str]] = Field(
+        None,
+        description='nodeSelector defines the nodeSelector to be supplied to Restic podSpec',
+    )
+    resourceAllocations: Optional[Any] = Field(
+        None,
+        description='resourceAllocations defines the CPU and Memory resource allocations for the restic Pod',
+    )
+    tolerations: Optional[List[TolerationModel]] = Field(
+        None,
+        description='tolerations defines the list of tolerations to be applied to Restic daemonset',
+    )
+
+
+class DataProtectionApplicationConfigurationVelero(BaseModel):
```

Comment:
please rename DataProtectionApplicationConfigurationVelero -> VeleroConfig

Reply:
There is this VeleroConfig
https://github.com/vmware-tanzu/velero/blob/v1.9.2/pkg/client/config.go#L38
which is different.
wdyt?
@mperetzred

Reply:
@amastbau you should refer at the DPA API, not the Velero's:
https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#VeleroConfig